### DISCUSSION TITLE
Are we planning to leave babel?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#730] [Q&A] [answered]
Solid use babel as the parser. It works with most current solutions, but not optimized for the next generations, **esbuild**, **swc**, etc.
Is there any plan for a more native solution to compile solid without babel to support modern frontend stacks that don't need babel anymore?

**Comment by atk** [comment, marked as answer]
Unfortunately, neither currently supports the dom-expression-babel-plugin that Solid requires to transform JSX. Currently, all eyes are on the plugin API redesign of swc, but we don't expect this to happen any time soon, since priorities are currently on SSR and solid-start.
